assets/basic/shaders/PbrMetallicRoughness.hlsl

```diff
@@ -0,0 +1,148 @@
+SamplerState              NormalMapSampler      : register(s4);
+Texture2D                 MetalRoughness        : register(t5);
+SamplerState              MetalRoughnessSampler : register(s6);
+
+struct VSOutput {
+  float4 world_position : POSITION;
+  float4 position       : SV_POSITION;
+  float2 uv             : TEXCOORD;
+  float3 normal         : NORMAL;
+  float3 normalTS    : NORMALTS;
+  float3 tangentTS   : TANGENTTS;
+  float3 bitangentTS : BITANGENTTS;
+};
+
+VSOutput vsmain(
+    float4 position : POSITION,
+    float2 uv       : TEXCOORD,
+    float3 normal   : NORMAL,
+    float3 tangent   : TANGENT)
+{
+  VSOutput result;
+
+  result.world_position = mul(Scene.ModelMatrix, position);
+  result.position = mul(Scene.CameraViewProjectionMatrix, result.world_position);
+  result.uv = uv;
+  result.normal = mul(Scene.ModelMatrix, float4(normal, 0)).xyz;
+  result.normalTS    = mul(Scene.ModelMatrix, float4(normal, 0)).xyz;
+  result.tangentTS   = mul(Scene.ModelMatrix, float4(tangent, 0)).xyz;
+  result.bitangentTS = cross(normal, tangent);
+
+  return result;
+}
+
+#define PI 3.1415292
+
+//
+// GGX/Towbridge-Reitz normal distribution function with
+// Disney's reparametrization of alpha = roughness^2
+//
+float DistributionGGX(float cosLh, float roughness)
+{
+    const float alpha   = roughness * roughness;
+    const float alphaSq = alpha * alpha;
+    const float denom   = (cosLh * cosLh) * (alphaSq - 1.0) + 1.0;
+    return alphaSq / (PI * denom * denom);
+}
+
+//
+// Single term for separable Schlick-GGX below
+//
+float GASchlickG1(float cosTheta, float k)
+{
+    return cosTheta / (cosTheta * (1.0 - k) + k);
+}
+
+//
+// Schlick-GGX approximation of geometric attenuation function using Smith's method
+//
+float GASmithSchlickGGX(float cosLi, float cosLo, float roughness)
+{
+    const float r = roughness + 1.0;
+    const float k = (r * r) / 8.0; // Epic suggests using this roughness remapping for analytic lights
+    return GASchlickG1(cosLi, k) * GASchlickG1(cosLo, k);
+}
+
+float3 FresnelSchlick(float3 F0, float cosTheta)
+{
+    return F0 + (1.0 - F0) * (float3)pow(1.0 - cosTheta, 5.0);
+}
+
+float4 psmain(VSOutput input) : SV_TARGET
+{
+    const float3   nTS = normalize(input.normalTS);
+    const float3   tTS = normalize(input.tangentTS);
+    const float3   bTS = normalize(input.bitangentTS);
+    const float3x3 TBN = float3x3(tTS.x, bTS.x, nTS.x,
+                            tTS.y, bTS.y, nTS.y,
+                            tTS.z, bTS.z, nTS.z);
+
+    const float3 V = normalize(Scene.EyePosition.xyz - input.world_position.xyz);
+    const float3 normal = NormalMap.Sample(NormalMapSampler, input.uv).rgb;
+    const float3 N = normalize(mul(TBN, normal * 2.0 - 1.0));
+
+    const float4 albedo = AlbedoTexture.Sample(AlbedoSampler, input.uv).rgba;
+    if (albedo.a < 0.8f) {
+      discard;
+    }
+    const float roughness = MetalRoughness.Sample(MetalRoughnessSampler, input.uv).g;
```

Comment:
No, the green one. The difference between the altimeter texture and the sponza scene is the red channel being either all-white, or all-black. But on both cases, green is roughness, blue metalness.
(looking at the asset is looks like red is used, but it's just because of the default to 1.0)